allennlp/training/initializers.py

```diff
@@ -0,0 +1,117 @@
+from typing import Dict, Callable
+import logging
+import re
+
+import torch.nn.init
+
+from allennlp.common.params import Params
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class InitializerApplicator:
+    """
+    Applies initializers to the parameters of a Module based on regex matches.
+    All parameters in the Module will be initialized.
+
+    """
+    def __init__(self, all_initializers: Dict[str, Callable[[torch.Tensor], None]] = None) -> None:
+        """
+        Parameters
+        ----------
+        all_initializers : Dict[str, Callable[[torch.Tensor], None]], optional (default = None)
+            A dictionary mapping parameter regexes to initializers to be applied to parameters
+            matching the regex. Additionally, it may include a "default" field containing a
+            initializer, which will be used in the case that the Applicator encounters a parameter
+            which does not match any of the regexes provided.
+
+        """
+        self._initializers = all_initializers or {}
```

Comment:
Why use `all_initializers` as the parameter name instead of just `initializers`?  You're just using `_initializers` for the member variable.